tests/test_templatetags.py

```diff
@@ -998,6 +1000,52 @@ def test_super_under_if_node(self):
             """,
         )
 
+    def test_nested_fills(self):
```

Comment:
Added a test to ensure the variable assigned via `default` is accessible also in nested components. And did the same for the `data` attribute too.